### DISCUSSION TITLE
Action no longer being triggered by `workflow_run`

### DISCUSSION THREAD

**Discussion by pljhll** [community/community#66887] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi all,

Running into a strange issue today where one of our workflows is no longer triggering when another successfully completes. This has been working fine for almost a year and last worked just fine 3 days ago. Suddenly, nothing. It's driving us a little nuts :)

I can't see any updates about this behaviour changing or any outages on the status page. Has anyone experienced something like this before?

Here's the workflow's trigger config:

```
on:
  workflow_run:
    workflows: [Deploy to QA]
    types: [completed]
```

Edit 12-SEP-23@16:55
We've got the exact same trigger type on another repo and just tested that and it works fine. So this doesn't seem to be a generic GitHub issue, maybe something related to our repo settings somewhere?

Edit 12-SEP-23@17:48
Several contributors experiencing the same. Thank you to everyone adding their info, hopefully we'll get to the bottom of it soon. It's coming to the end of our day here so I'll pick the thread back up in the morning.

Cheers,

Paul.

**Comment by DaMandal0rian** [comment]
I'm having an issue with self hosted runners, no iobs are triggering. Last successful run was 8 hrs ago, then the latest runner image release v2.309 was automatically downloaded and applied, and then all jobs fail.

**Reply by pljhll** [reply]
Thanks for the info, these are github hosted runners (I should have mentioned that) and other jobs are triggering on merges etc just fine.

The mystery continues...

**Reply by DaMandal0rian** [reply]
Thank you, did you problem start today?

**Reply by pljhll** [reply]
It did start today, yes.

**Comment by rogerbramon** [comment]
We are experiencing this issue. The [only difference compared to the documentation](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-branches-for-workflow-run-events) seems the use of "" (e.g. "Deploy to QA" instead of Deploy to QA). I'll give it a try.

**Reply by pljhll** [reply]
That's interesting! We have it working without the use of "" in another repo just fine and this has been okay until today, but I'd be interested to know if you can reproduce this.

**Comment by tschaub** [comment]
I'm seeing the same.  When using `types: [completed]`, the workflow is never triggered.  I do see that it can be triggered by the `in_progress` and `requested` types.  But not `completed`.  Here is a workflow that I am testing with https://github.com/openlayers/bench/actions/workflows/check.yml

**Reply by pljhll** [reply]
Interesting, my action was in the format I posted above and has worked just fine until today.

**Reply by pljhll** [reply]
We've seen completed triggering correctly in another repo with the same yaml. I don't think it's a related thread, but @rogerbramon is looking at the use of "" in the job name for the trigger.

**Reply by rogerbramon** [reply]
I added the check.yml workflow from @tschaub to a repo and it's working fine for me (without brackets). So weird

**Comment by ivan-sre** [comment]
Probably my issue is related. I have my GitHub App subscribed on "Workflow run" events. It stopped to get "workflow_run.completed" event.

**Reply by pljhll** [reply]
Hey Ivan, did your issue start today as well?

**Reply by ivan-sre** [reply]
No, we  have this problem several days already

**Reply by ivan-sre** [reply]
Actually, it really started today.

**Comment by andrewyoo** [comment]
Experiencing the same thing trying to use `workflow_run` and `completed`. It was working yesterday and not today.

```
name: Deploy Production
on:
  workflow_run:
    workflows: [ Rails ]
    types: [ completed ]
    branches: [ main ]
concurrency: production_environment
```

**Reply by pljhll** [reply]
Thanks for confirming @andrewyoo. Do you have any other repos that use this setup? We do and they are working fine, so it seems localised.

**Reply by andrewyoo** [reply]
No, this is the only one where i'm using `workflow_run`. All other workflows triggered by other `on`s appear to be working fine. This one appears to not be triggered at all as i don't see it in the Actions tab as being attempted even.

**Comment by suhas-coursera** [comment]
We're facing the same issue in the below workflow from yesterday eve:

> name: FE Apps CD
> 
> on:
>   workflow_run:
>     workflows: [FE Apps Build]
>     types: [completed]
>     branches: [main]


But the other workflow configured to run via workflow_run is running fine even today, not sure if it's related to quotes or branch, testing further.

name: Automated Alerts
> on:
>   workflow_run:
>     workflows: ["Upload data to S3"]
>     types:
>       - completed

**Reply by andrewyoo** [reply]
Could it be the syntax now?

```
types: [completed]
```

vs
```
types:
- completed
```

It seems like all 3 broken workflows are using `types: [completed]`?

**Reply by pljhll** [reply]
I definitely have a working example using `types: [completed]` here. It's identical in syntax to the one that's failing in another repo...

**Reply by suhas-coursera** [reply]
No luck by adding the quotes or removing the branch condition. so it seems a problem on the GitHub side.

**Reply by rogerbramon** [reply]
Should we try to involve someone from the GitHub staff? Perhaps @JamesMGreene may help?

**Reply by JamesMGreene** [reply]
I've raised the flag internally to see if anyone is already looking at this. 👀

**Reply by JamesMGreene** [reply]
A recent change has been rolled back now. Feel free to ping again if you're still experiencing issues with future workflow runs, starting now.

**Reply by andrewyoo** [reply]
confirmed working for me again! thanks!

**Comment by tschaub** [comment]
I've also tried all variety of syntax tweaks.  Here is a workflow example that I would expect to run on `completed`, `requested`, and `in_progress` type `workflow_run` events: https://github.com/openlayers/bench/blob/2b4bb752aecf6959796d8e017112ede9864b012f/.github/workflows/check.yml.  Instead, the workflow is only being run on `requested` and `in_progress`.  So no more `completed` type runs.

**Reply by pljhll** [reply]
That's very useful, thanks @tschaub ! I haven't found any release notes from GitHub that suggest a change here.

The only thing that doesn't fit the pattern is that we do have a working `completed` trigger in a workflow in another repo that doesn't seem affected. I can't see any differences though.

**Reply by andrewyoo** [reply]
Thanks for testing! I submitted a bug report. Given we're all having the same problem today, i'm guessing this is a github issue.

**Reply by suhas-coursera** [reply]
Agree with @paul-kuihi, I too have workflow triggering with `completed` but in a different repo, so unclear of the root cause.

**Comment by CatalinScaesteanu** [comment]
Hey, I'm having the same issue.
Since Friday, 7th of August. All our workflows that should trigger after other workflows have completed have stopped working.

Any news on this?

**Comment by JamesMGreene** [comment, marked as answer]
A recent change (~24 hours) has been rolled back now.

Feel free to ping again if you're still experiencing issues with future workflow runs, starting now.

**Reply by CatalinScaesteanu** [reply]
It's working now for us. 
Thank you!

**Reply by pljhll** [reply]
Amazing, thank you @JamesMGreene and everyone else that helped with this.

**Reply by PradipFopse** [reply]
Hello @JamesMGreene ,

We are facing a similar issue. My colleague has started a new thread for the issue in https://github.com/orgs/community/discussions/67197
Could you please review the thread and help? This is a critical issue for our project.

Regards
Pradip
